### PR TITLE
Update Push Notifications

### PR DESCRIPTION
**Pull request by ConnorLand**
I will be dropping docs here from Dropbox Paper drafts, updating them, and validating them.

If a Dropbox Paper doc has a red "Closed Doc" warning at the top, please make your changes here in GitHub instead.

### CODE REVIEW

File: content/howto/mobile/notif-implement-native.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+1. Open up your project navigation.
+2. Go to your Native mobile navigation profile
+3. Click Synchronization configuration
+4. Add `DeviceRegistration` and `NativeNotification` entities from the `PushNotifications` module
+5. Set the `Download` column for those two new entities to synchronize to `All Objects`
+
+This will ensure that the correct objects are synchronized to your native apps.
+
+Now that you have everything setup it is time to deploy your native app. Click here for instructions on enabling push notifications when building a native app.
```
Comment (MitchelMx) [inline content/howto/mobile/notif-implement-native.md:51]: "Click here" should refer to notif-build-native

File: content/howto/mobile/notif-implement-module.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	c. `User` is for any user role that needs to interact with notifications without being an `Administrator` or `Anonymous` user role.
+1. Save the Security settings.
+1. Open up **Project** > **Navigation** in the App Explorer.
+1. Add a new open page navigation item to the `Responsive` navigation profile.
+1. Select the page **Administration** from the **PushNotifications** module.
+	a. This page is located in the `_Use Me/Web` folder of the **PusNotifications** module.
+1. Add a microflow sub-call to your **Afterstartup** microflow for the microflow **AfterStartup_PushNotifications**.
+	a. If you do not have any afterstartup microflow configured yet read here how to set one up.
+
+Now you are able to start your application and move on to the next step configuring your runtime.
```
Comment (MitchelMx) [inline content/howto/mobile/notif-implement-module.md:40]: "configuring your runtime" should refer to notif-config-push

File: content/howto/mobile/notif-config-push.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+| GoogleServices-Info.plist | Google Firebase | Used to push notifications to your iOS native applications.     |
+| google-services.json      | Google Firebase | Used to push notifications to your Android native applications. |
+
+These files are also known as private key files. They must be configured as you build your native apps for Android and iOS, and also on the runtime when configuring your server. This last part is done from the administration page and setup wizard.
+
+If you did follow the instructions on the page to set up your apps for iOS or Android in for the server sided part.
+
+After doing that you will be redirected to the configuration page. For more information on the configuration page click here.
+
+Now that we have set up the backend part it is time to implement hybrid/native push notifications for your hybrid/native app. Click here for hybrid or here for native.
```
Comment (MitchelMx) [inline content/howto/mobile/notif-config-push.md:36]: or "here" for native should refer to notif-implement-native

File: content/howto/mobile/notif-build-native.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+1. After choosing the type of build (local development or distribution) go to `App capabilities`
+1. Under `Firebase configuration` switch `Push notifications` to **On**
+1.  Scroll down and upload the firebase configurations
+    a. `google-services.json` for the Android build
+    b. `GoogleServices-Info.plist` for the iOS build
+1. Save the configuration and now you are ready to build
+
+When building for local development, keep in mind that the `Make it Native 8/9` apps from Mendix do not support your own push notifications. This means you will always have to build your own local package and distribute it to an emulator (Android only)/test device for testing native push notifications.
+
+Now you are able to build, after making sure your application runs in an emulator/test device click here for instructions on sending your first test notification.
```
Comment (MitchelMx) [inline content/howto/mobile/notif-build-native.md:29]: Click "here" for... should refer to notif-send-test

File: content/howto/mobile/notif-build-native.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+menu_order: 60
+description: Tutorial for building a native app with push notifications enabled.
+tags: ["mobile", "native", "push", "local", "firebase"]
+---
+
+## 1 Introduction
+
+Now that your application is ready to deploy to iOS and/or Android it is time to build the native mobile app.
+
+[ ] Explain why
```
Comment (MitchelMx) [inline content/howto/mobile/notif-build-native.md:13]: Could be replaced by the following text:

Deploying your app to iOS and/or Android is required to interact with the mobile device that is required to receive the push notifications from the server.

File: content/howto/mobile/notifications.md [hunk 6]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
 tags: ["mobile", "native", "push", "local", "firebase"]
 ---
 
+## 1 Introduction {#intro}
+
 These guides will teach you to configure both push notifications (also known as remote notifications) and local notifications for your Mendix apps:
 
-Push Notifications:
+* [Add Module Dependencies](add-module-depends) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:13]: Explains installing required dependency modules.

File: content/howto/mobile/notifications.md [hunk 7]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
...
+* [Implement the Push Notifications Module](notif-implement-module) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:14]: Shows how to implement the push notification module itself.

File: content/howto/mobile/notifications.md [hunk 8]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
...
+* [Configure Push Notifications](notif-config-push) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:15]: Guides to configure push notifications on runtime.

File: content/howto/mobile/notifications.md [hunk 9]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
...
+* [Native Push Notification Implementation](notif-implement-native) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:16]: Gives a step by step guide on how to implement notifications for native applications.

File: content/howto/mobile/notifications.md [hunk 10]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
...
+* [Build a Native Mobile App with Push Notifications](notif-build-native) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:17]: Helps you build a native app with push notifications enabled.

File: content/howto/mobile/notifications.md [hunk 11]
```diff
@@ -6,13 +6,30 @@ description: Tutorials for configuring push and local notifications.
...
+* [Send Your First Test Push Notification](notif-send-test) — 1 sentence doc description.
```
Comment (MitchelMx) [inline content/howto/mobile/notifications.md:18]: The final step in validating your push notification implementation.

**Comment by ConnorLand** [comment]
Need to merge dev into this branch and then confirm these changes are inside it: https://github.com/mendix/docs/pull/3659

File: content/howto/mobile/notif-implement-module.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+1. Open the Push Notification Connector module.
+1. Click **Download**.
+
+## 3 Configure Your App {#config}
+
+In order for your app to use to use the Push Connections Module, configure the following things:
+
+1. Open **Project** > **Security** in the App Explorer.
+1. Go to **User roles**.
+1.  Adjust the user roles that should have access to the push notifications:
```
Comment (ConnorLand) [inline content/howto/mobile/notif-implement-module.md:28]: Do these requirements sync up with the existing format of Prototype/demo or Production? Perhaps this portion of the doc is already accomplished by the way Mendix's Prototype/demo or Production security levels come out of the box? @MitchelMx

File: content/howto/mobile/notif-implement-module.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+	a. `Administrator` module role should be assigned to at least one user executing configuration and administrative tasks. <br />
+	b. `Anonymous` module role may be assigned to your anonymous project user role. <br />
+	c. `User` is for any user role that needs to interact with notifications without being an `Administrator` or `Anonymous` user role.
+1. Save the Security settings.
+1. Open up **Project** > **Navigation** in the App Explorer.
+1. Add a new open page navigation item to the `Responsive` navigation profile.
```
Comment (ConnorLand) [inline content/howto/mobile/notif-implement-module.md:34]: I think open page is "show a page" now. I'll confirm and then change.

Comment (ConnorLand) [reply content/howto/mobile/notif-implement-module.md:34]: Also I will tell the user to just type in "administration" when searching in the pages. It makes things come up quickly.

File: content/howto/mobile/notif-implement-module.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+1. Select the page **Administration** from the **PushNotifications** module.
+	a. This page is located in the `_Use Me/Web` folder of the **PusNotifications** module.
+1. Add a microflow sub-call to your **Afterstartup** microflow for the microflow **AfterStartup_PushNotifications**.
```
Comment (ConnorLand) [inline content/howto/mobile/notif-implement-module.md:37]: Will change "sub-call" to "At the end of your **Afterstartup** microflow, add a Microflow call activity at the end and select **AfterStartup_PushNotifications**."

File: content/howto/mobile/notif-implement-module.md (hunk 12 above)
Comment (MitchelMx) [reply content/howto/mobile/notif-implement-module.md:28]: I do not fully get what you mean by this comment @ConnorLand  could you elaborate on it a bit more?

File: content/howto/mobile/notif-implement-module.md (hunk 13 above)
Comment (MitchelMx) [reply content/howto/mobile/notif-implement-module.md:34]: As long as they know which page exactly to get as well that should be fine.

File: content/howto/mobile/notif-config-push.md [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+
+## 2 Configure Your Notifications
+
+If you have completed [Add Module Dependencies](add-module-depends) and [Implement the Push Notifications Module](notif-implement-module) per your use case, do the following to configure your push notifications:
+
+1. Open your app in Mendix Studio Pro.
+1. Log in as the Administrator user you [previously designated](notif-implement-module#config).
+1. Navigate to the **Administration** page.
+
+The first time you open this page it will present you with a wizard to set up Firebase configurations. These are required to use push notifications. If you have not set up Firebase configurations, see [doc title](nodoc). The wizard will mention the following files:
```
[the remark this answers is not in the document]
Comment (MitchelMx) [reply content/howto/mobile/notif-config-push.md:23]: Correct, they need files from Firebase before they can execute the first time setup in the application.

File: content/howto/mobile/notif-build-native.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+
+## 2 Build Your Native App
+
+1. In Studio Pro top bar navigation
+1.  Go to Project/App > Build Native Mobile App
+    a. If you are building your native app for the first time click here for instructions
+1. After choosing the type of build (local development or distribution) go to `App capabilities`
+1. Under `Firebase configuration` switch `Push notifications` to **On**
+1.  Scroll down and upload the firebase configurations
```
Comment (MitchelMx) [inline content/howto/mobile/notif-build-native.md:22]: Here we will only be enabling iOS and/or Android.
+ upload the private key generated as part of this step in the FCM setup https://docs.mendix.com/howto/mobile/setting-up-google-firebase-cloud-messaging-server#setting-up-a-service-account
+ we only need to enter the project number once.

**Comment by ConnorLand** [comment]
> @ConnorLand what's the outlook on publishing this and closing this PR?

I will be finishing most of the docs work this week. I think we should be able to merge early next week. I just need to check if the Push Notifications Connector module is ready to publish. It has been throwing some strange errors the last few days.

Regardless, I will work to make the docs ready so that when the module is error-free, we are good to publish.

File: content/howto/mobile/notif-add-module-depends.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+## 4 Implement the Nanoflow Commons Module
+
+Skip this section if the [Nanoflow Commons](https://marketplace.mendix.com/link/component/109515/Mendix/Nanoflow-Commons) module is already implemented in your app, or if you do not plan on using using local or native push notifications. Implement this module by doing the following:
+
+1. Open up the Marketplace from Studio Pro.
+1. Search for *Nanoflow commons*.
+1. Open the [Nanoflow commons](https://marketplace.mendix.com/link/component/109515/Mendix/Nanoflow-Commons) module.
+1. Click **Download**.
+1. No further installation is required, if you want to know more about the module visit the [Nanoflow Commons marketplace guide](https://docs.mendix.com/appstore/modules/nanoflow-commons).
```
[the remark this answers is not in the document]
Comment (ConnorLand) [reply content/howto/mobile/notif-add-module-depends.md]: Correction made.

File: content/howto/mobile/notif-build-native.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+title: "Build a Native App with Push Notifications Enabled"
+parent: "notifications"
+menu_order: 60
+description: Tutorial for building a native app with push notifications enabled.
+tags: ["mobile", "native", "push", "local", "firebase"]
+---
+
+## 1 Introduction
+
+Now that your application is ready to deploy to iOS and/or Android it is time to build the native mobile app.
```
[the remark this answers is not in the document]
Comment (ConnorLand) [reply content/howto/mobile/notif-build-native.md]: Understood. Could you take a look at the new intro text @KevinVlaanderen ?

File: content/howto/mobile/notif-build-native.md (hunk 5 above)
Comment (ConnorLand) [reply content/howto/mobile/notif-build-native.md]: Used this as inspiration for the new intro text.

File: content/howto/mobile/notif-build-native.md [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+[ ] Explain why
+
+## 2 Build Your Native App
+
+1. In Studio Pro top bar navigation
+1.  Go to Project/App > Build Native Mobile App
+    a. If you are building your native app for the first time click here for instructions
```
[the remark this answers is not in the document]
Comment (ConnorLand) [reply content/howto/mobile/notif-build-native.md:17]: You're right. I've put in the comma, and more corrections to this doc are coming today.

**Comment by ConnorLand** [comment]
I need to upload a cloned cluster of new Push Notification docs into the Mx8 docs. They need to have a note at the top requiring this prerec:

* To use Push Notifications in Mx8, please upgrade to 8.18 (the long-term support version of Mx8).

I will also be sure to update UI text with Mx8 appropriate text (changing App -> Project maybe).

**Comment by ConnorLand** [comment]
> I need to upload a cloned cluster of new Push Notification docs into the Mx8 docs. They need to have a note at the top requiring this prerec:
> 
> * To use Push Notifications in Mx8, please upgrade to 8.18 (the long-term support version of Mx8).
> 
> I will also be sure to update UI text with Mx8 appropriate text (changing App -> Project maybe).

These changes will be done in a new Mx8 PR that I will make so we can merge that documentation earlier.

**Pull request merged**